### PR TITLE
Use CUDA_PATH on Windows to find Cuda

### PR DESCRIPTION
**Pull request by AndrewGaspar**
This change to the build script searches in `CUDA_PATH` for Cuda when targeting Windows when `CUDA_LIBRARY_PATH` is not set. `CUDA_PATH` is set in the typical Cuda install on Windows, so this change makes it easier to use `cuda-sys` and derived crates on Windows, without having to do anything beyond installing Cuda.

I tried to support building for 32-bit Windows. Nvidia does not ship 32-bit builds for `cublas.lib`, which `cuda-sys` currently requires. The code is still there, though, in case this requirement is relaxed.

### CODE REVIEW

**Comment by AndrewGaspar** [comment]
The "10.0-devel-ubuntu16.04" run looks like noise. Could you re-run it?

**Comment by AndrewGaspar** [comment]
Hey @bheisler, do you have any feedback on this change?

**Comment by bheisler** [comment]
Seems reasonable to me. The CUDA_PATH variable is set on my machine.

**Comment by AndrewGaspar** [comment]
Oh, heh, sorry. I had assumed you were the maintainer of this repo.

**Comment by bheisler** [comment]
Ah, no, I don't have write access so I can't merge this pull request. I believe @termoshtt would have access.

File: build.rs
Comment (termoshtt) [inline build.rs]: Could you use `cfg!(target_os = "windows")` and `cfg!(target_arch = x86_64)` instead of parsing `TARGET` environment ?

**Review by termoshtt — APPROVED** [review]
Thank PR. I have confirmed this works on my windows machine (with test failures about `long`)

File: build.rs
Comment (AndrewGaspar) [reply build.rs]: A blunt `cfg!(target_os = "windows")` for the whole CUDA_PATH branch is probably fine, but we need to read `TARGET` to support cross-compiling for 32-bit Windows (and WoW) from the 64-bit Windows since build scripts are always compiled for the native architecture, even when cross-compiling.

Comment (AndrewGaspar) [reply build.rs]: But cross-compiling is otherwise currently blocked by #9 (Nvidia does not ship 32-bit libraries for cublas)

Comment (AndrewGaspar) [reply build.rs]: I added a check for `cfg!(target_os = "windows")` before checking for `CUDA_PATH`.

**Comment by termoshtt** [comment]
:+1:

**Pull request merged**
